### PR TITLE
Serialize and deserialize Connecting

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-sdk/lib/core/services/WalletService.ts [hunk 1]
```diff
@@ -166,6 +167,7 @@ export class WalletService {
     const {privateKey, securityCode} = await this.sdk.connect(contractAddress);
     const connectingWallet: ConnectingWallet = new ConnectingWallet(contractAddress, name, privateKey);
     this.setConnecting(connectingWallet);
+    this.setState({kind: 'Connecting', wallet: connectingWallet});
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/services/WalletService.ts]: You can deduplicate some code, by using `initializeConnection` here

File: universal-login-wallet/src/ui/react/ConnectAccount/ConnectAccount.tsx [hunk 2]
```diff
@@ -6,11 +6,12 @@ import {useHistory} from 'react-router';
...
 export type ConnectModal = 'connectionFlow' | 'selector';
 
-export const ConnectAccount = () => {
+export const ConnectAccount = ({walletState}: {walletState: any}) => {
   const {sdk, walletService} = useServices();
   const history = useHistory();
   const [name, setName] = useState<string | undefined>(undefined);
-  const [connectModal, setConnectModal] = useState<ConnectModal>('selector');
+  const defaultModal = walletState.kind === 'Connecting' ? 'connectionFlow' : 'selector';
+  const [connectModal, setConnectModal] = useState<ConnectModal>(defaultModal);
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/ConnectAccount/ConnectAccount.tsx]: Why not make it a route?

File: universal-login-sdk/lib/core/services/WalletService.ts (hunk 1 above)
Comment (marekkirejczyk) [reply universal-login-sdk/lib/core/services/WalletService.ts]: Perhaps you can use `waitForConnection` below

File: universal-login-react/src/ui/ConnectionFlow/ConnectionFlow.tsx [hunk 3]
```diff
@@ -21,10 +21,17 @@ export interface ConnectionFlowProps {
 
 export const ConnectionFlow = ({basePath = '', name, onCancel, onSuccess, sdk, walletService, className}: ConnectionFlowProps) => {
   const history = useHistory();
+
+  const onConnectWithDeviceClick = () => {
+    walletService.initializeConnection(name).then(() => {
+      history.push('/connect/emoji');
+    });
```
Comment (xitronix) [inline universal-login-react/src/ui/ConnectionFlow/ConnectionFlow.tsx]: It is not important, but i would like:
walletService.initializeConnection(name).then(() => history.push('/connect/emoji'));

File: universal-login-wallet/src/ui/react/ConnectAccount/ConnectAccount.tsx (hunk 2 above)
Comment (xitronix) [reply universal-login-wallet/src/ui/react/ConnectAccount/ConnectAccount.tsx]: PR #1656

**Pull request merged**